pkg/kms/localkms/crypto_box.go

```diff
@@ -0,0 +1,251 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package localkms
+
+import (
+	"bytes"
+	"crypto/ed25519"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/tink/go/aead"
+	"github.com/google/tink/go/keyset"
+	ed25519pb "github.com/google/tink/go/proto/ed25519_go_proto"
+	tinkpb "github.com/google/tink/go/proto/tink_go_proto"
+	"golang.org/x/crypto/nacl/box"
+
+	"github.com/hyperledger/aries-framework-go/pkg/internal/cryptoutil"
+	"github.com/hyperledger/aries-framework-go/pkg/kms"
+	"github.com/hyperledger/aries-framework-go/pkg/kms/localkms/internal/keywrapper"
+	"github.com/hyperledger/aries-framework-go/pkg/secretlock/noop"
+)
+
+// TODO: move CryptoBox out of the KMS package.
+//   this currently only sits inside LocalKMS so it can access private keys. See issue #511
+// TODO delete this file and its corresponding test file when LegacyPacker is removed.
+
+// CryptoBox provides an elliptic-curve-based authenticated encryption scheme
+//
+// Payloads are encrypted using symmetric encryption (XChacha20Poly1305)
+// using a shared key derived from a shared secret created by
+//   Curve25519 Elliptic Curve Diffie-Hellman key exchange.
+//
+// CryptoBox is created by a LegacyKMS, and reads secret keys from the LegacyKMS
+//   for encryption/decryption, so clients do not need to see
+//   the secrets themselves.
+type CryptoBox struct {
+	km *LocalKMS
+}
+
+// NewCryptoBox creates a CryptoBox which provides crypto box encryption using the given KMS's key.
+func NewCryptoBox(w kms.KeyManager) (*CryptoBox, error) {
```

Comment:
why not receive *LocalKMS structure instead of checking it inside?

Reply:
it can be *LocalKMS... not spending too much thinking about this CryptoBox functionality as it's tied to LegacyPacker and may be removed in the future. But by default a constructor uses interface arguments and implementations can be checked internally, this is what was followed here. 

I guess this argument can be changed to be the implementation type instead because only LegacyPacker uses it.

Reply:
We can't change kms.KeyManager to *LocalKMS since this is called in the packager that calls this via a provider. If we do change, then this conversion will move up to the packager code. It's better to do it here rather than duplicating the conversion for each use of NewCryptoBox() even if packager is the only user.